frontend/src/types.ts

```diff
@@ -1473,3 +1476,48 @@ export interface Breadcrumb {
     /** Whether to show a custom popup */
     popup?: Pick<PopupProps, 'overlay' | 'sameWidth' | 'actionable'>
 }
+
+export enum GraphTypes {
+    Bar = 'bar',
+    HorizontalBar = 'horizontalBar',
+    Line = 'line',
+    Histogram = 'histogram',
+    Pie = 'doughnut',
+}
+
+export type GraphDataset = ChartDataset<ChartType> &
+    Partial<
+        Pick<
+            TrendResult,
+            | 'count'
+            | 'label'
+            | 'days'
+            | 'labels'
+            | 'data'
+            | 'compare'
+            | 'status'
+            | 'action'
+            | 'actions'
+            | 'breakdown_value'
+        >
+    > & {
+        id: number
+        dotted?: boolean
+        borderDash?: number[]
+        breakdownValues?: string[]
+        persons_urls?: { url: string }[]
+    }
```

Comment:
Can you add some comments to describe what each part of this complex type does and where does it come from?